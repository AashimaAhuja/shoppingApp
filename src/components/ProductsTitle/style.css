.products-title {
  display: flex;
  justify-content: space-between;
  margin: 8px;
}

.products-title > button {
  padding: 8px;
}
