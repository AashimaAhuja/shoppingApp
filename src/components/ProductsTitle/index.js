import React from "react";
import "./style.css";

const ProductsTitle = ({ productsCount, onToggleSoldItems, showSoldItems }) => {
  return (
    <div className="products-title">
      <span>{productsCount} Results</span>
      <button onClick={onToggleSoldItems}>
        {showSoldItems ? "Hide Sold Items" : "Show Sold Items"}
      </button>
    </div>
  );
};
export default ProductsTitle;
