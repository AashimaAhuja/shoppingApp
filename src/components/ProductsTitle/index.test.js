import { shallow } from 'enzyme';
import ProductsTitle from './';

const defaultProps = {
  productsCount: 1,
  onToggleSoldItems: jest.fn(),
  showSoldItems: false
}

describe('<ProductsTitle/> Tests', ()=>{
  it('should render ProductsTitle Component', ()=>{
    const ProductsTitleComponent = shallow(<ProductsTitle />)
    expect(ProductsTitleComponent).toMatchSnapshot();
  });

  it('should handle ProductsTitle button click', ()=>{
    const mockFn = jest.fn();
    const wrapper = shallow(<ProductsTitle {...defaultProps} onToggleSoldItems={mockFn} />)
    expect(wrapper).toMatchSnapshot();
    const button = wrapper.find('button');
    button.simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });

  it('should handle ProductsTitle when showSoldItems is true', ()=>{
    const wrapper = shallow(<ProductsTitle {...defaultProps} showSoldItems={true} />)
    expect(wrapper).toMatchSnapshot();
    const button = wrapper.find('button');
    expect(button.text()).toEqual('Hide Sold Items');
  });
  
  it('should handle ProductsTitle when showSoldItems is false', ()=>{
    const wrapper = shallow(<ProductsTitle {...defaultProps} />)
    expect(wrapper).toMatchSnapshot();
    const button = wrapper.find('button');
    expect(button.text()).toEqual('Show Sold Items');
  });
})
