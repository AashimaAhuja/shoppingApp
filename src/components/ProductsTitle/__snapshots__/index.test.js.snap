// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductsTitle/> Tests should handle ProductsTitle button click 1`] = `
<div
  className="products-title"
>
  <span>
    1
     Results
  </span>
  <button
    onClick={[MockFunction]}
  >
    Show Sold Items
  </button>
</div>
`;

exports[`<ProductsTitle/> Tests should handle ProductsTitle when showSoldItems is false 1`] = `
<div
  className="products-title"
>
  <span>
    1
     Results
  </span>
  <button
    onClick={[MockFunction]}
  >
    Show Sold Items
  </button>
</div>
`;

exports[`<ProductsTitle/> Tests should handle ProductsTitle when showSoldItems is true 1`] = `
<div
  className="products-title"
>
  <span>
    1
     Results
  </span>
  <button
    onClick={[MockFunction]}
  >
    Hide Sold Items
  </button>
</div>
`;

exports[`<ProductsTitle/> Tests should render ProductsTitle Component 1`] = `
<div
  className="products-title"
>
  <span>
     Results
  </span>
  <button>
    Show Sold Items
  </button>
</div>
`;
